parser/parse.go

```diff
@@ -463,7 +463,14 @@ func (b *Builder) findTypesIn(pkgPath string, u *types.Universe) error {
 		tf, ok := obj.(*tc.Func)
 		// We only care about functions, not concrete/abstract methods.
 		if ok && tf.Type() != nil && tf.Type().(*tc.Signature).Recv() == nil {
-			b.addFunction(*u, nil, tf)
+			t := b.addFunction(*u, nil, tf)
+			c1 := b.priorCommentLines(obj.Pos(), 1)
+			t.CommentLines = splitLines(c1.Text())
+			if c1 == nil {
```

Comment:
is it kosher to call c1.Text() on the previous line if c1 is nil?  If you have a convincing argument, comment it.  else fix.